analysis/src/xml.rs

```diff
@@ -0,0 +1,715 @@
+        .filter(|node| !node.ancestors().any(|node| node.has_tag_name("comment")))
+        .map(|node| node.text().unwrap())
+        .collect::<String>()
+}
+
+/// Returns true when the `node`'s "api" attribute matches the `expected` API.
+fn api_matches(node: &Node, expected: &str) -> bool {
+    node.attribute("api")
+        .map(|values| values.split(',').any(|value| value == expected))
+        .unwrap_or(true)
+}
+
+/// Raw representation of Vulkan XML files (`vk.xml`, `video.xml`).
+#[derive(Debug, Default)]
+pub struct Registry {
+    pub externals: Vec<External>,
+    pub basetypes: Vec<BaseType>,
+    pub bitmask_types: Vec<BitMaskType>,
+    pub bitmask_aliases: Vec<Alias>,
+    pub handles: Vec<Handle>,
+    pub handle_aliases: Vec<Alias>,
+    pub enum_types: Vec<EnumType>,
+    pub enum_aliases: Vec<Alias>,
+    pub funcpointers: Vec<FuncPointer>,
+    pub structs: Vec<Structure>,
+    pub struct_aliases: Vec<Alias>,
+    pub unions: Vec<Structure>,
+    pub constants: Vec<Constant>,
+    pub constant_aliases: Vec<Alias>,
+    pub enums: Vec<Enum>,
+    pub bitmasks: Vec<BitMask>,
+    pub commands: Vec<Command>,
+    pub command_aliases: Vec<Alias>,
+    pub features: Vec<Feature>,
+    pub extensions: Vec<Extension>,
+}
+
+impl Registry {
+    pub fn parse(input: &'static str, api: &str) -> Registry {
+        let doc = roxmltree::Document::parse(input).unwrap();
+        Registry::from_node(doc.root_element(), api)
+    }
+
+    fn from_node(registry_node: Node, api: &str) -> Registry {
+        let mut registry = Registry::default();
+        for registry_child in registry_node
+            .children()
+            .filter(|node| api_matches(node, api))
+        {
+            match registry_child.tag_name().name() {
+                "types" => {
+                    for type_node in registry_child
+                        .children()
+                        .filter(|node| node.has_tag_name("type"))
+                        .filter(|node| api_matches(node, api))
+                    {
+                        if type_node.has_attribute("alias") {
+                            match type_node.attribute("category") {
+                                Some("bitmask") => {
+                                    registry.bitmask_aliases.push(Alias::from_node(type_node));
+                                }
+                                Some("handle") => {
+                                    registry.handle_aliases.push(Alias::from_node(type_node));
+                                }
+                                Some("enum") => {
+                                    registry.enum_aliases.push(Alias::from_node(type_node));
+                                }
+                                Some("struct") => {
+                                    registry.struct_aliases.push(Alias::from_node(type_node));
+                                }
+                                _ => (),
```

Comment:
Warn about new/unhandled items? Same below.

Reply:
I designed this parser around it only filtering out data WE want, and ignoring everything else. If new nodes with an unknown category get added, and we want to use that data, we can easily expand this. It's not often that there are new types of data points which are of use to our generator, and adding checks everywhere throughout the code to look for these feels unnecessary in my opinion.

Reply:
> though a `debug!()` stating all the features that we haven't implemented (for sensible reasons) might still be nice

Implemented this. I have also set up `env_logger` to go along with this.

Reply:
> Implemented this. I have also set up `env_logger` to go along with this.

Hm, CI fails because `env_logger` 0.11, which came out just hours ago, has an MSRV of 1.71. It's probably not worth bumping our MSRV over this? So, is it fine depending on `env_logger` 0.10? Or maybe use another logging library?

Reply:
Switched to `tracing` now, which is more modern, and has a lower MSRV than `env_logger`.